File: src/DigiBouquetAILinks.php
<?php

declare(strict_types=1);

namespace DigiBouquetAI\SiteKit;

final class DigiBouquetAILinks
{
    public const BASE_URL = 'https://digibouquetai.com';

    public static function homeUrl(): string
    {
        return self::BASE_URL;
    }

    public static function pageUrl(string $slug = ''): string
    {
        $clean = trim($slug, '/');
        return $clean === '' ? self::BASE_URL : self::BASE_URL . '/' . $clean;
    }

    public static function createBouquetUrl(): string
    {
        return self::pageUrl('create-bouquet');
    }

    public static function aiGiftCardUrl(): string
    {
        return self::pageUrl('ai-card');
    }

    public static function examplesUrl(): string
    {
        return self::pageUrl('bouquet-examples');
    }

    public static function guidesUrl(): string
    {
        return self::pageUrl('guides');
    }

    public static function pricingUrl(): string
    {
        return self::pageUrl('pricing');
    }

    public static function aboutUrl(): string
    {
        return self::pageUrl('about');
    }
}

File: src/main/java/com/digibouquetai/sitekit/DigiBouquetAILinks.java
package com.digibouquetai.sitekit;

/**
 * URL helpers for DigiBouquet AI.
 */
public final class DigiBouquetAILinks {
    /** DigiBouquet AI homepage. */
    public static final String BASE_URL = "https://digibouquetai.com";

    private DigiBouquetAILinks() {
    }

    /**
     * Returns the DigiBouquet AI homepage URL.
     *
     * @return homepage URL
     */
    public static String homeUrl() {
        return BASE_URL;
    }

    /**
     * Builds a DigiBouquet AI page URL from a path slug.
     *
     * @param slug page path slug
     * @return page URL
     */
    public static String pageUrl(String slug) {
        String clean = slug == null ? "" : slug.replaceAll("^/+|/+$", "");
        return clean.isEmpty() ? BASE_URL : BASE_URL + "/" + clean;
    }

    /** Returns the bouquet builder URL.
     * @return bouquet builder URL
     */
    public static String createBouquetUrl() {
        return pageUrl("create-bouquet");
    }

    /** Returns the AI gift-card URL.
     * @return AI gift-card URL
     */
    public static String aiGiftCardUrl() {
        return pageUrl("ai-card");
    }

    /** Returns the bouquet examples URL.
     * @return bouquet examples URL
     */
    public static String examplesUrl() {
        return pageUrl("bouquet-examples");
    }

    /** Returns the guides URL.
     * @return guides URL
     */
    public static String guidesUrl() {
        return pageUrl("guides");
    }

    /** Returns the pricing URL.
     * @return pricing URL
     */
    public static String pricingUrl() {
        return pageUrl("pricing");
    }

    /** Returns the about URL.
     * @return about URL
     */
    public static String aboutUrl() {
        return pageUrl("about");
    }
}
